src/App.tsx

```diff
@@ -1,73 +1,108 @@
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+interface AutocompleteProps {
+  delay?: number;
+  onSelected: (person: Person | null) => void;
+}
 
-  return (
-    <div className="container">
-      <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
+const Autocomplete: React.FC<AutocompleteProps> = ({ delay = 300, onSelected }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [suggestions, setSuggestions] = useState<Person[]>([]);
+  const [debouncedValue, setDebouncedValue] = useState(inputValue);
 
-        <div className="dropdown is-active">
-          <div className="dropdown-trigger">
-            <input
-              type="text"
-              placeholder="Enter a part of the name"
-              className="input"
-              data-cy="search-input"
-            />
-          </div>
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(inputValue);
+    }, delay);
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [inputValue, delay]);
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
+  useEffect(() => {
+    if (debouncedValue === '') {
+      setSuggestions(peopleFromServer);
+    } else {
+      const filteredSuggestions = peopleFromServer.filter(person =>
+        person.name.toLowerCase().includes(debouncedValue.toLowerCase())
+      );
+      setSuggestions(filteredSuggestions);
+    }
+  }, [debouncedValue]);
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setInputValue(value);
+    if (value === '') {
+      onSelected(null); // Очистити вибрану особу, коли введення порожнє
+    } else {
+      onSelected(null); // Очистити вибрану особу, коли введення змінюється
+    }
```

Comment:
The `onSelected(null)` call is redundant in both branches of the if-else statement. You can remove the else block and keep only the `onSelected(null)` call after setting the input value.